### ISSUE TITLE
Intermittent UI test failure - SmokeTest.editCustomSearchEngineTest

### ISSUE THREAD

**Issue by sv-ohorvath** [mozilla-mobile/fenix#21540]
### Firebase Test Run:
https://console.firebase.google.com/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/8266266295243895427/executions/bs.3645e7ee27e73c92/testcases/1/test-cases
### Stacktrace:
`java.lang.IllegalArgumentException: Can't scroll to index 5, it is out of bounds [0, 0)
at androidx.compose.foundation.lazy.LazySemanticsKt$lazyListSemantics$1$3.invoke(LazySemantics.kt:78)
at androidx.compose.foundation.lazy.LazySemanticsKt$lazyListSemantics$1$3.invoke(LazySemantics.kt:77)
at androidx.compose.ui.test.ActionsKt$performScrollToIndex$2.invoke(Actions.kt:145)
at androidx.compose.ui.test.ActionsKt$performScrollToIndex$2.invoke(Actions.kt:144)
at androidx.compose.ui.test.junit4.AndroidSynchronization_androidKt$sam$java_util_concurrent_Callable$0.call(Unknown Source:2)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:458)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at android.app.Instrumentation$SyncRunnable.run(Instrumentation.java:2163)
at android.os.Handler.handleCallback(Handler.java:873)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:193)
at android.app.ActivityThread.main(ActivityThread.java:6669)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)`
### Build: 
Main 9/28

**Comment by AndiAJ** [comment]
Flaky 1x on 9/29 Main [Firebase link](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/6811040629745959438/executions/bs.eafdb77766e1d92e/testcases/1/test-cases)

**Comment by AndiAJ** [comment]
Fixed with #21591

**Comment by Archaeopteryx** [comment]
Still observed for 0d9e2b3a3604924dea1c3d3b47b480ad0591318f ([log](https://console.firebase.google.com/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/8774589772551172664/executions/bs.f3846e026c7e0883/test-cases)).

**Comment by sv-ohorvath** [comment]
Flaky on 10/1: [logs](https://console.firebase.google.com/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/8774589772551172664/executions/bs.f3846e026c7e0883/testcases/1/test-cases)

new stacktrace:
`java.lang.AssertionError
at org.junit.Assert.fail(Assert.java:87)
at org.junit.Assert.assertTrue(Assert.java:42)
at org.junit.Assert.assertTrue(Assert.java:53)
at org.mozilla.fenix.ui.robots.SettingsSubMenuSearchRobot.saveNewSearchEngine(SettingsSubMenuSearchRobot.kt:61)
at org.mozilla.fenix.ui.SmokeTest$editCustomSearchEngineTest$4.invoke(SmokeTest.kt:565)
at org.mozilla.fenix.ui.SmokeTest$editCustomSearchEngineTest$4.invoke(SmokeTest.kt:561)
at org.mozilla.fenix.ui.robots.SettingsRobot$Transition.openSearchSubMenu(SettingsRobot.kt:128)
at org.mozilla.fenix.ui.SmokeTest.editCustomSearchEngineTest(SmokeTest.kt:561)`

**Comment by AndiAJ** [comment]
Fixed with #21803

**Comment by AaronMT** [comment]
```Kotlin
java.lang.AssertionError
at org.junit.Assert.fail(Assert.java:87)
at org.junit.Assert.assertTrue(Assert.java:42)
at org.junit.Assert.assertTrue(Assert.java:53)
at org.mozilla.fenix.ui.robots.SettingsSubMenuSearchRobot.saveNewSearchEngine(SettingsSubMenuSearchRobot.kt:60)
at org.mozilla.fenix.ui.SmokeTest$editCustomSearchEngineTest$4.invoke(SmokeTest.kt:563)
at org.mozilla.fenix.ui.SmokeTest$editCustomSearchEngineTest$4.invoke(SmokeTest.kt:559)
at org.mozilla.fenix.ui.robots.SettingsRobot$Transition.openSearchSubMenu(SettingsRobot.kt:128)
at org.mozilla.fenix.ui.SmokeTest.editCustomSearchEngineTest(SmokeTest.kt:559)
```
